include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp

```diff
@@ -136,43 +154,38 @@ class alignment_executor_two_way
      * \tparam exec_policy_t The type of the execution policy; seqan3::is_execution_policy must return `true`. Defaults
      *                       to seqan3::sequenced_policy.
      *
-     * \param[in] resrc The underlying resource containing the sequence pairs to align.
+     * \param[in] resource The underlying resource containing the sequence pairs to align.
      * \param[in] fn The alignment kernel to invoke on the sequences pairs.
-     * \param[in] buffer_value A value to initialise the buffer with.
-     * \param[in] chunk_size The number of sequence pairs to invoke the alignment algorithm with.
+     * \param[in] buffer_value A dummy object to deduce the type of the underlying buffer value.
      * \param[in] exec Optional execution policy to use. Defaults to seqan3::seq.
      *
      * \throws std::invalid_argument if the chunk size is less than 1.
      *
      * \details
      *
-     * Forwards the resource range as a zipped view with an index view to provide internal ids for the alignments.
      * If the execution handler is parallel, it allocates a buffer of the size of the given resource range.
      * Otherwise the buffer size is 1.
+     * Also note that the third argument is used for deducing the type of the underlying buffer value and is otherwise
+     * not used in the context of the class' construction.
      */
     template <typename exec_policy_t = sequenced_policy>
     //!\cond
         requires is_execution_policy_v<exec_policy_t>
     //!\endcond
-    alignment_executor_two_way(resource_t resrc,
+    alignment_executor_two_way(resource_t resource,
                                alignment_algorithm_t fn,
-                               value_t const buffer_value = value_t{},
-                               size_t chunk_size = 1u,
+                               value_t const SEQAN3_DOXYGEN_ONLY(buffer_value) = value_t{},
```

Comment:
Does the buffer_value type differ from the result type of the algorithm?

Reply:
No it should not. This is only required to offer a type deduction guide. Unfortunately, I can't partially specialise the templates and use CTAD. But in the propsal that @marehr linked this would go away as the buffer and the execution model woud be more cleanly separated.

Reply:
Why is the `value_t` a template parameter of the two_way_executor if  the `alignment_algorithm_t` type already is, can't you use the `function_traits` you introduced recently to access the value_type of the algorithm? Or do you mean a different type deduction?

Reply:
No, the problem was that before we knew that we handle alignment algorithms and we knew the exact structure of the corresponding function object. We might not know such thing about any other algorithm. Thus, to decouple it from the algorithm (which it really does not need to know in this interface) the expected buffer value type must be given from outside. And in the above mebtioned proposal it will be even better encapsulated as we will have a buffer implementation which does know nothing about the execution and an executor that does know nothing about the buffer, only that it needs to have a certain interface so they can interact with each other. So the buffer is then similar to a vector, just with a dedicated interface to handle all the things it needs to handle the results of the algorithm and what else we need. And the executor only knows the algorithm to be called  the resource to be called on and the somehow the buffer slot it has to pass to the algorithm so it can fill it